book/specifications/kimchi/template.md

```diff
@@ -1,16 +1,87 @@
 # Kimchi
 
-This document specifies the kimchi variant of PLONK.
+* This document specifies *kimchi*, a zero-knowledge proof system that's a variant of PLONK.
+* This document does not specify how circuits are created, or executed. Only how to convert a circuit and its execution into a proof.
 
 ## Overview
 
-TKTK
+There's three main algorithms to kimchi:
+
+* [Setup](#constraint-system-creation): takes a circuit and produces a prover index, and a verifier index.
+* [Proof creation](#proof-creation): takes the prover index, and the execution trace of the circuit to produce a proof.
+* [Proof verification](#proof-verification): takes the verifier index and a proof to verify.
+
+As part of these algorithms, a number of tables are created (and then converted into polynomials) to create a proof.
+
+**Gates**. A circuit is described by a series of gates, that we list in a table. 
+The columns of the tables list the gates, while the rows are the length of the circuit. 
+For each row, only a single gate can take a value $1$ while all other gates take the value $0$.
+
+|  row  | Generic | Poseidon | CompleteAdd | VarBaseMul | EndoMul | EndoMulScalar | ChaCha0 | ChaCha1 | ChaCha2 | ChaChaFinal |
+| :---: | :-----: | :------: | :---------: | :--------: | :-----: | :-----------: | :-----: | :-----: | :-----: | :---------: |
+|   0   |    1    |    0     |      0      |     0      |    0    |       0       |    0    |    0    |    0    |      0      |
+|   1   |    0    |    1     |      0      |     0      |    0    |       0       |    0    |    0    |    0    |      0      |
+
+**Coefficients**. The coefficient table has 15 columns, and is used to tweak the gates. 
```

Comment:
> Is the number of columns of the coefficients table necessarily linked to the number of columns of the registers?

Indirectly. Right it is linked to the Poseidon gate, which needs 15 round constants to be specified somewhere (and different poseidon gate might have different round constants, so it's not fixed). In turn, the poseidon manages to do 15 transitions because we have 15 registers/columns.